image-toolkit/sbin/startup.sh

```diff
@@ -67,7 +67,19 @@ webcam_stream_loop() {
                 vcodec=h264,acodec=none,vb=3000,ab=0,fps=15 \
             }:duplicate{ \
                 dst=std{access=rtmp,mux=ffmpeg{mux=flv},dst=rtmp://localhost/live/webcam}, \
-            }"
+            }" &
+        CLVC_PID=$!
+        # Monitor the process and device
+        while [[ -e "/dev/video$DEVICE_NO" ]] && ps -p $CLVC_PID > /dev/null; do
```

Comment:
You should monitor /dev/video with inotifywait (for example, `inotifywait -e delete_self /dev/video0` will exit when /dev/video0 is disconnected) and `wait -n <pidlist>` (wait for any in <pidlist> to finish).

Monitoring /dev/video0 by repeatedly polling could miss quick reconnection events (for example, a contestant accidentally touching the webcam's wire).

Reply:
Thanks for your suggestion. I will update the code accordingly.